source/FreeRTOS_DHCPv6.c

```diff
@@ -445,7 +448,7 @@ static BaseType_t xDHCPv6ProcessEndPoint_HandleAdvertise( NetworkEndPoint_t * px
 
     #if ( ipconfigUSE_DHCP_HOOK != 0 )
         /* Ask the user if a DHCP request is required. */
-        eAnswer = xApplicationDHCPHook( eDHCPPhasePreRequest, EP_DHCPData.ulOfferedIPAddress );
+        eAnswer = xApplicationDHCPHook_IPv6( eDHCPPhasePreRequest, pxEndPoint, &( pxDHCPMessage->xIPAddress ) );

```

Comment:
Thanks Hein for the changes. 
I have one more feedback. It will be good if we can use the same DHCP Hook for IPv4 and IPv6 (xApplicationDHCPHook). We will just need to use the IP_Address_t structure which has both v4 and v6 addresses.